tests/spec_decode/test_spec_decode_worker.py

```diff
@@ -215,13 +222,20 @@ def test_correctly_calls_rejection_sampler(k: int, batch_size: int):
         actual.target_probs,
         target_token_probs.reshape(batch_size, k + 1, -1)[:, :-1])
     assert torch.equal(actual.draft_token_ids, proposal_token_ids)
-    assert torch.equal(actual.draft_probs, proposal_probs)
+    if isinstance(spec_decode_sampler, RejectionSampler):
+        # The draft probabilites is used only by the RejectionSampler.
+        # Ensure that if the sampler is a RejectionSampler then the
+        # draft probs are being passed.
+        assert torch.equal(actual.draft_probs, proposal_probs)
```

Comment:
I missed this in the other PR. Can we have both acceptance routines take in the draft probabilities so we don't have branches like this? We can have the typical acceptance just ignore them.

Reply:
Fixed this. Added the same interface for both the samplers. I have included the changes in this PR because it is not a lot of changes. Let me know if that is fine for you in terms of reviewing.